crates/asm/subprotocols/bridge-v1/src/state/bridge.rs

```diff
@@ -0,0 +1,897 @@
+        tx: &Transaction,
+        info: &DepositInfo,
+    ) -> Result<(), DepositValidationError> {
+        // Validate the deposit first
+        self.validate_deposit(tx, info)?;
+
+        let notary_operators = self.operators().current_multisig_indices();
+        let entry = DepositEntry::new(info.deposit_idx, info.outpoint, notary_operators, info.amt)?;
+        self.deposits.insert_deposit(entry)?;
+
+        Ok(())
+    }
+
+    /// Validates a deposit transaction and info against bridge state requirements.
+    ///
+    /// This function performs comprehensive validation of a deposit by verifying:
+    /// - The deposit amount matches the bridge's expected amount
+    /// - The Deposit Request Transaction (DRT) spending signature is valid
+    /// - The deposit output is properly locked to the aggregated operator key
+    /// - The deposit index is unique within the deposits table
+    ///
+    /// # Parameters
+    ///
+    /// - `tx` - The Bitcoin transaction containing the deposit
+    /// - `info` - The parsed deposit information to validate
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(())` - If the deposit passes all validation checks
+    /// - `Err(DepositError)` - If validation fails for any reason
+    ///
+    /// # Errors
+    ///
+    /// Returns error if:
+    /// - The deposit amount doesn't match the bridge's expected amount
+    /// - The DRT spending signature is invalid or doesn't match the aggregated operator key
+    /// - The deposit output lock is incorrect
+    /// - A deposit with the same index already exists
+    fn validate_deposit(
+        &self,
+        tx: &Transaction,
+        info: &DepositInfo,
+    ) -> Result<(), DepositValidationError> {
+        // Verify the deposit amount matches the bridge's expected amount
+        if info.amt.to_sat() != self.denomination.to_sat() {
+            return Err(DepositValidationError::MismatchDepositAmount(Mismatch {
+                expected: self.denomination.to_sat(),
+                got: info.amt.to_sat(),
+            }));
+        }
+
+        // Validate the DRT spending signature against the aggregated operator key
+        validate_drt_spending_signature(
+            tx,
+            info.drt_tapscript_merkle_root,
+            self.operators().agg_key(),
+            info.amt.into(),
+        )?;
+
+        // Ensure the deposit output is properly locked to the aggregated operator key
+        validate_deposit_output_lock(tx, self.operators().agg_key())?;
+
+        // Verify this deposit index hasn't been used before
+        if self.deposits().get_deposit(info.deposit_idx).is_some() {
+            return Err(DepositValidationError::DepositIdxAlreadyExists(
+                info.deposit_idx,
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// Creates a withdrawal assignment by selecting an unassigned deposit and assigning it to an
+    /// operator.
+    ///
+    /// This function handles incoming withdrawal commands by:
```

Comment:
Is "withdrawal command" different than withdrawal request? Any reason to use this terminology (over the term we used so far)?